pkg/agent/handlers/response.go

```diff
@@ -0,0 +1,47 @@
+package handlers
```

Comment:
would it be better to have some errors wrapped
e.g. 
```
errors.Wrap(err, errors.NotFound)

```

and next the mapper which will check what kinds of error this is and set valid response